### DISCUSSION TITLE
Logrotate error

### DISCUSSION THREAD

**Discussion by Cris70** [nextcloud/all-in-one#2848] [Questions] [answered]
### Steps to reproduce
1. setup log rotation for container logs
2. see journalctl of logrotate service

### Expected behavior 
Container log should be rotated

### Actual behavior 
logrotate gives access denied error and cannot rotate

### Host OS 
Rocky Linux 9.2

#### Nextcloud AIO version 
6.1.1

#### Current channel 
latest

#### Other valuable info 
I have logrotate service set up to rotate container logs daily.
I do not remember if this step was outlined in the setup guide of Nextcloud AIO, or if I found it somewhere else, or if it was an idea of mine. However, I have this logrtoate configuration in place:
```
/var/lib/docker/containers/*/*.log {
  rotate 7
  daily
  compress
  missingok
  delaycompress
  copytruncate
}
```
I never noticed it in the past, but now I see that since about 10 days logrotate is giving errors like these:
```
Jun 23 00:00:15 nextcloud-vm.codispace.local logrotate[2791964]: error: stat of /var/lib/docker/containers/2dcf7d17700cc547d82a2e1f2c1c439f3584160436ead3482a80c84ca93bf326/2dcf7d17700cc547d82a2e1f2c1c439f3584160436ead3482a80c84ca93bf326-json.log failed: Permission denied
Jun 23 00:00:15 nextcloud-vm.codispace.local logrotate[2791964]: error: stat of /var/lib/docker/containers/362ff155f4691f5ce1d12fe3a2914143221ccbd6bc9e5f54606c0394fcf5d813/362ff155f4691f5ce1d12fe3a2914143221ccbd6bc9e5f54606c0394fcf5d813-json.log failed: Permission denied
Jun 23 00:00:15 nextcloud-vm.codispace.local logrotate[2791964]: error: stat of /var/lib/docker/containers/39484bd81d756b48272824551d809a02cf424b8cda95f764b628851e240e7adf/39484bd81d756b48272824551d809a02cf424b8cda95f764b628851e240e7adf-json.log failed: Permission denied
Jun 23 00:00:15 nextcloud-vm.codispace.local logrotate[2791964]: error: stat of /var/lib/docker/containers/42586aaaa62c2e36061c3fe86fc279708e8a76438fedc382c73aeeb158f9bc61/42586aaaa62c2e36061c3fe86fc279708e8a76438fedc382c73aeeb158f9bc61-json.log failed: Permission denied
Jun 23 00:00:15 nextcloud-vm.codispace.local logrotate[2791964]: error: stat of /var/lib/docker/containers/5221dc3da779bd268956477fc867bb3db8560092e2d2c6db9e53e1cdecae8dc9/5221dc3da779bd268956477fc867bb3db8560092e2d2c6db9e53e1cdecae8dc9-json.log failed: Permission denied
Jun 23 00:00:15 nextcloud-vm.codispace.local logrotate[2791964]: error: stat of /var/lib/docker/containers/60e7095ca82c0ea58ed13676cbe3d3f873e50c9a708b995d2a23124d4d14cae6/60e7095ca82c0ea58ed13676cbe3d3f873e50c9a708b995d2a23124d4d14cae6-json.log failed: Permission denied
Jun 23 00:00:15 nextcloud-vm.codispace.local logrotate[2791964]: error: stat of /var/lib/docker/containers/63233d5f244508598c7cf3d6d3c57a5f27579dfeb99b58f60ee797f94335f419/63233d5f244508598c7cf3d6d3c57a5f27579dfeb99b58f60ee797f94335f419-json.log failed: Permission denied
Jun 23 00:00:15 nextcloud-vm.codispace.local logrotate[2791964]: error: stat of /var/lib/docker/containers/a74248c71c45ae7da498dc6296ea6bc93d0bbaf6416aefbfa50f69b208755465/a74248c71c45ae7da498dc6296ea6bc93d0bbaf6416aefbfa50f69b208755465-json.log failed: Permission denied
Jun 23 00:00:15 nextcloud-vm.codispace.local logrotate[2791964]: error: stat of /var/lib/docker/containers/b246f57dcd2afce12280c12f13bde7f211443957e58984faf6872f0446adb234/b246f57dcd2afce12280c12f13bde7f211443957e58984faf6872f0446adb234-json.log failed: Permission denied
Jun 23 00:00:15 nextcloud-vm.codispace.local logrotate[2791964]: error: stat of /var/lib/docker/containers/d4729a700a7b5f8810b8122506dfde804841756a2597ab888195ec3be9ca735f/d4729a700a7b5f8810b8122506dfde804841756a2597ab888195ec3be9ca735f-json.log failed: Permission denied
Jun 23 00:00:15 nextcloud-vm.codispace.local logrotate[2791964]: error: stat of /var/lib/docker/containers/ea4783cd9dd59786826610177c7966a100fe8ae97f2c50f5a7afcc2c1602f2b8/ea4783cd9dd59786826610177c7966a100fe8ae97f2c50f5a7afcc2c1602f2b8-json.log failed: Permission denied
```
Is it supposed to work? Or is it something I made up during a sleepless night and completely wrong?
If the latter, is there something in place to avoid for container logs to grow indefinitely?

Thank you in advance

**Comment by Zoey2936** [comment]
Thats seems not like a AIO issue, more like a docker/logrotate problem, does logrotate run as root?

**Comment by Cris70** [comment]
Thank you @Zoey2936 for your reply.
Just tried it now. If launched from the CLI as root, it works.
I also found where I got that setup from. It's written [here](https://sandro-keil.de/blog/logrotate-for-docker-container/), and it is referenced in the Nextcloud AIO installation instructions.

**Comment by joshtrichards** [comment]
Try this approach instead: 

https://docs.docker.com/config/containers/logging/json-file/

That's the Docker method (I haven't tested it with All-In-One though).

That linked blog post is extremely old. Though it has a note that Docker >1.8 users - which was like 8 years ago at this point hah - can use the above approach.

You're right - that blog post is linked from the AIO installation instructions here:

https://github.com/nextcloud/all-in-one#huge-docker-logs

If the official Docker approach works well with AIO we should probably change that link to go directly to the official Docker docs.

Maybe it's even worthwhile seeing if there is a reasonable way for us to provide default rotation settings for the AIO containers.

**Comment by szaimen** [comment, marked as answer]
We switched almost all container logs to warn mode which should not produce that much logs anymore so that logrotate is not needed anymore at least for AIO containers. I removed the instructions with https://github.com/nextcloud/all-in-one/commit/d0cae686365409daab32c71b962a7debd8e5262c

**Comment by Cris70** [comment]
Thank you everyone for the help and the suggestions.

**Comment by Zoey2936** [comment]
maybe we could add the log-opts max-size option https://docs.docker.com/config/containers/logging/local to the container.json file?

**Reply by szaimen** [reply]
This is appatlrently configured in daemon.json which we dont have access to, no?

**Reply by Zoey2936** [reply]
it can be set per continaer by using `--log-driver local --log-opt max-size=10m --log-opt max-file=3 --log-opt compress=true` or something like this, alternative we could link this documentation file in the readme

**Reply by szaimen** [reply]
Ah I see. Then it is indeed possible. However I'd say due to adjusting the loglevel to warn this is not needed anymore? Also it can be adjusted in daemon.json for all containers apparently...

**Reply by Zoey2936** [reply]
> Ah I see. Then it is indeed possible. However I'd say due to adjusting the loglevel to warn this is not needed anymore? Also it can be adjusted in daemon.json for all containers apparently...

then maybe mention this document at the place where logrotate was mentioned before?

**Reply by szaimen** [reply]
https://github.com/nextcloud/all-in-one/pull/2852
